scapy/packet.py

```diff
@@ -1407,6 +1407,56 @@ def command(self):
             c += "/" + pc
         return c
 
+    def convert_to(self, other_cls, **kwargs):
+        """Converts this Packet to another type.
+
+        This is not guaranteed to be a lossless process.
+
+        By default, this only implements conversion to ``Raw``.
+
+        :param other_cls: Reference to a Packet class to convert to.
+        :returns: Converted form of the packet, or None if no conversion is
+                  available.
+        """
+        if not issubtype(other_cls, Packet):
+            raise TypeError("{} must implement Packet".format(other_cls))
+
+        if other_cls is Raw:
+            return Raw(raw(self))
+
+        if self.haslayer(other_cls):
+            return self.getlayer(other_cls)
+
+        if "_internal" not in kwargs:
+            return other_cls.convert_packet(self, _internal=True, **kwargs)
+
```

Comment:
Also I'd rather raise an error if the conversion isn't implemented

Reply:
Done.